test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextNoTrackingStreamsTests.cs

```diff
@@ -0,0 +1,450 @@
+    ""value"":[
+            {""Id"":1,""Name"":""U1""},
+            {""Id"":2,""Name"":""U2""},
+            {""Id"":3,""Name"":""U3""}
+        ]
+}
+";
+        private string CUSTOMERS_RESPONSE = @"{
+  ""@odata.context"": ""https://localhost:8000/$metadata#Customers"",
+  ""value"": [
+    {
+      ""@odata.ProfilePhotoReadLink"": ""https://localhost:8000/Customers/1/ProfilePhoto"",
+      ""@odata.ProfilePhotoContentType"": ""image/jpg"",
+      ""Id"": 1,
+      ""Name"": ""Sample Doc 1"",
+      ""FileLength"": 0
+    }
+  ]
+}";
+
+
+        private const string DOCUMENTS_RESPONSE = @"{
+  ""@odata.context"": ""https://localhost:8000/$metadata#Documents"",
+  ""value"": [
+    {
+      ""@odata.mediaReadLink"": ""https://localhost:8000/Documents/1/content"",
+      ""@odata.mediaContentType"": ""text/plain"",
+      ""Id"": 1,
+      ""Name"": ""Sample Doc 1"",
+      ""FileLength"": 0
+    },
+   {
+      ""@odata.mediaReadLink"": ""https://localhost:8000/Documents/2/content"",
+      ""@odata.mediaContentType"": ""text/plain"",
+      ""Id"": 2,
+      ""Name"": ""Sample Doc 2"",
+      ""FileLength"": 0
+    },
+    {
+      ""@odata.mediaReadLink"": ""https://localhost:8000/Documents/3/content"",
+      ""@odata.mediaContentType"": ""text/plain"",
+      ""Id"": 3,
+      ""Name"": ""Sample Doc 3"",
+      ""FileLength"": 0
+    }
+  ]
+}";
+
+        private const string DOCUMENT_RESPONSE = @"{
+      ""@odata.context"": ""https://localhost:8000/$metadata#Documents/$entity"",
+      ""@odata.mediaReadLink"": ""https://localhost:8000/Documents/1/content"",
+      ""@odata.mediaContentType"": ""text/plain"",
+      ""Id"": 1,
+      ""Name"": ""Sample Doc 1"",
+      ""FileLength"": 0
+    }";
+
+        private const string USER_RESPONSE = @"{
+    ""@odata.context"":""http://localhost:8000/$metadata#Users/$entity"",
+    ""Id"":1,
+    ""Name"":""U1""
+      }";
+
+        #endregion
+        private const string AttachmentResponse = "Hello World!";
+
+        public DataServiceContextNoTrackingStreamsTests()
+        {
+            var uri = new Uri("http://localhost:8000");
+
+            NonTrackingContext = new Container(uri)
+            {
+                MergeOption = MergeOption.NoTracking
+            };
+
+            DefaultTrackingContext = new Container(uri);
+        }
+
+        [Fact]
+        public void TestBehaviourShouldRemainTheSameForTheDataContext()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { DefaultTrackingContext, NonTrackingContext }, false);
+
+            var users = DefaultTrackingContext.Users.ExecuteAsync().GetAwaiter().GetResult();
+            var users2 = NonTrackingContext.Users.ExecuteAsync().GetAwaiter().GetResult();
+            // data should be fetched and the count should be the same
+            Assert.Equal(users.ToList().Count, users2.ToList().Count);
+
+            //  entities should be tracked by no tracking context

```

Comment:
Did you mean "Entities should **not** be tracked by no-tracking context"?

Reply:
for addition it seems that all entities are tracked. only fetched entities are not

Reply:
But in this case you're testing fetching, not addition. Even your assertions test that the no-tracking context has 0 entities in its entity tracker:

```c#
Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.ToList().Count);
```

Reply:
Fixed